src/test/java/org/assertj/core/api/objectarray/ObjectArrayAssert_flatExtractingEasy_Test.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2015 the original author or authors.
+ */
+package org.assertj.core.api.objectarray;
+
+import org.assertj.core.api.iterable.Extractor;
+import org.assertj.core.test.CartoonCharacter;
+import org.assertj.core.test.ExpectedException;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ObjectArrayAssert_flatExtractingEasy_Test {
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  private CartoonCharacter bart;
+  private CartoonCharacter lisa;
+  private CartoonCharacter maggie;
+  private CartoonCharacter homer;
+  private CartoonCharacter pebbles;
+  private CartoonCharacter fred;
+
+  @Before
+  public void setUp() {
+    bart = new CartoonCharacter("Bart Simpson");
+    lisa = new CartoonCharacter("Lisa Simpson");
+    maggie = new CartoonCharacter("Maggie Simpson");
+
+    homer = new CartoonCharacter("Homer Simpson");
+    homer.getChildren().add(bart);
+    homer.getChildren().add(lisa);
+    homer.getChildren().add(maggie);
+
+    pebbles = new CartoonCharacter("Pebbles Flintstone");
+    fred = new CartoonCharacter("Fred Flintstone");
+    fred.getChildren().add(pebbles);
+  }
+
+  @Test
+  public void should_allow_assertions_on_joined_lists_when_extracting_children() {
+    assertThat(new CartoonCharacter[] { homer, fred }).flatExtracting("children").containsOnly(bart, lisa, maggie,
+        pebbles);
+  }
+
+  @Test
+  public void should_allow_assertions_on_empty_result_lists() throws Exception {
+    assertThat(new CartoonCharacter[] { bart, lisa, maggie }).flatExtracting("children").isEmpty();
+  }
```

Comment:
The NPE of the older version was related to the extractor implementation in the test class. With the new version an IllegalArgumentException is thrown. My first idea was to rethrow the IllegalArgumentException as a NPE but that doesn't look good. So, just a test for the IllegalArgumentException?

     private final Extractor<CartoonCharacter, List<CartoonCharacter>> children = new Extractor<CartoonCharacter, List<CartoonCharacter>>() {
        @Override
        public List<CartoonCharacter> extract(CartoonCharacter input) {
          return input.getChildren();
        }
      };